Import ssmtp-2.64-14.el7.src.rpm as part of XS-1557

glehmann opened the pull request:
* rebuild for openssl3
scratch build: https://koji.xcp-ng.org/taskinfo?taskID=89692



Update from @rzr

- [x] Koji build Scratch (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=94242
- [x] Koji pre-build XCPNG2710 (v8.3-u-pcoval2): https://koji.xcp-ng.org/taskinfo?taskID=94704

glehmann committed:
Import ssmtp-2.64-14.el7.src.rpm

Signed-off-by: Gaëtan Lehmann <gaetan.lehmann@gmail.com>

ydirson reviewed (changes requested):
XS seems to just add a build-fix patch and breq's, all drowned in their spec-rewrite. We may want to add the patches more cleany, or else keep original changelog and document what they did

stormi commented:
We're clearly at a turning point where most of what we don't like in XS 9 packaging is trying to contaminate more packages in XCP-ng 8.3. I wouldn't object to, and might even favour, going with the el7 packaging and adding necessary patches/breqs on top, while crediting XS for the work.

rzr commented on SPECS/ssmtp.spec:
```diff
@@ -0,0 +1,131 @@
+Patch5: ssmtp-validate-TLS-server-cert.patch
+Patch6: ssmtp-defaultvalues.patch
+Patch7: ssmtp-configure-c99.patch
+## Source0:	ftp://ftp.debian.org/debian/pool/main/s/%%{name}/%%{name}_%%{version}.orig.tar.bz2
```
Comment could be moved above Source0:  line

but this better to align to our upstream

stormi replied:
All these files are coming from one upstream and/or the other, indeed.

rzr commented on SPECS/ssmtp.spec:
```diff
@@ -0,0 +1,131 @@
+Requires(preun):	%{_sbindir}/alternatives
+BuildRequires: make
+BuildRequires:  gcc
+BuildRequires:	openssl-devel
```
is there a minimal version required ?

stormi replied:
Probably not, and unless we wan't to make our packaging differ from the one from XS, I think we can keep it as is without regrettable consequences.

rzr reviewed (commented):
May you please elaborate 
- rebuild for openssl3
?

It look like there are 2 imports in

https://github.com/xcp-ng-rpms/ssmtp/pull/1/commits

stormi commented:
The two imports are:
* the version of the el7 package that we believe XenServer has forked, to the upstream-8.3 branch
* the version of the XS package, to the XS-8.3 branch, based on upstream-8.3

rzr committed:
Update obsolete patch macro with autosetup

Like done in ssmtp-2.64-15.xs8.src.rpm
which contains unneeded "SOURCES/ssmtp-configure-c99.patch"

Using upstream base with autosetup does not require to patch generated
file stored in tarball, as it is regenerated on the fly (in %prep)

Origin: https://github.com/xcp-ng-rpms/ssmtp/pull/1
Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

rzr committed:
Add gcc to BuildRequires

Signed-off-by: Philippe Coval <philippe.coval@vates.tech>

rzr commented:
> XS seems to just add a build-fix patch and breq's, all drowned in their spec-rewrite. We may want to add the patches more cleany, or else keep original changelog and document what they did

This patch is not even needed with autoregen see my changes

glehmann commented:
Approved :heavy_check_mark:

rzr reviewed (approved):
Approving for @glehmann:

https://github.com/xcp-ng-rpms/ssmtp/pull/1#issuecomment-3877289738

rzr commented:
- [x] Koji build (v8.3-incoming): https://koji.xcp-ng.org/taskinfo?taskID=97974#ssmtp#89bbcb1
